File: skill_manager/application/skills/observations.py
from __future__ import annotations

from dataclasses import dataclass
from pathlib import Path

from .package import SkillPackage


@dataclass(frozen=True)
class SkillObservation:
    harness: str
    label: str
    scope: str
    package: SkillPackage


@dataclass(frozen=True)
class StorePackageObservation:
    package: SkillPackage
    recorded_revision: str | None = None
    recorded_source_ref: str | None = None
    recorded_source_path: str | None = None
    origin_harness: str | None = None


@dataclass(frozen=True)
class SkillsHarnessScan:
    harness: str
    label: str
    logo_key: str | None
    installed: bool
    skills: tuple[SkillObservation, ...] = ()
    excluded_skill_names: tuple[str, ...] = ()


@dataclass(frozen=True)
class SkillStoreScan:
    packages: tuple[StorePackageObservation, ...] = ()
    issues: tuple[str, ...] = ()


__all__ = [
    "SkillObservation",
    "SkillStoreScan",
    "SkillsHarnessScan",
    "StorePackageObservation",
]

File: src-tauri/src/skills/observations.rs
use std::path::PathBuf;

use super::identity::SourceDescriptor;
use super::package::SkillPackage;

#[derive(Debug, Clone)]
pub struct SkillObservation {
    pub harness: String,
    pub label: String,
    pub scope: String,
    pub package: SkillPackage,
}

#[derive(Debug, Clone)]
pub struct StorePackageObservation {
    pub package: SkillPackage,
    pub recorded_revision: Option<String>,
    pub recorded_source_ref: Option<String>,
    pub recorded_source_path: Option<String>,
    pub origin_harness: Option<String>,
}

#[derive(Debug, Clone)]
pub struct SkillsHarnessScan {
    pub harness: String,
    pub label: String,
    pub logo_key: Option<String>,
    pub installed: bool,
    pub skills: Vec<SkillObservation>,
    pub excluded_skill_names: Vec<String>,
}

#[derive(Debug, Clone)]
pub struct SkillStoreScan {
    pub packages: Vec<StorePackageObservation>,
    pub issues: Vec<String>,
}

#[derive(Debug, Clone)]
pub struct InventorySighting {
    pub kind: String,
    pub harness: Option<String>,
    pub label: String,
    pub scope: Option<String>,
    pub path: Option<PathBuf>,
    pub revision: Option<String>,
    pub source: SourceDescriptor,
    pub detail: String,
}
